AxonDeepSeg/data_management/input_data.py

```diff
@@ -1,437 +1,131 @@
-
-            image, real_mask, real_weights = self.apply_data_augmentation([image, real_mask, real_weights],
-                                                                          augmented_data, data_aug_verbose)
+        data_gen_args = dict(vertical_flip=True)
+        image_datagen = ImageDataGenerator(**data_gen_args)
+        mask_datagen = ImageDataGenerator(**data_gen_args)
 
-            # Normalisation of the image
-            image = apply_legacy_preprocess(image)
+        seed = 2018
 
-            # We have now loaded the good image, a mask (under the shape of a matrix, with different labels) that still needs to be converted to a volume (meaning, a sparse cube where each layer of depth relates to a class)
 
-            batch_x.append(image)
-            batch_y.append(real_mask)
-            batch_w.append(real_weights)
+        image_generator = image_datagen.flow(self.__load__(id_name)[0], seed=seed, batch_size= batch_size, shuffle=True)
+        mask_generator = mask_datagen.flow(self.__load__(id_name)[1], seed=seed, batch_size= batch_size, shuffle=True)
 
-            # If we are at the end of an epoch, we reset the list of samples, so that during next epoch all sets will be different.
-            if self.sample_seen == self.epoch_size:
-                if each_sample_once:
-                    self.samples_list = self.reset_set(type_ = 'validation')
-                    break
-                else:
-                    self.samples_list = self.reset_set(type_ = 'train')
-
-
-        # Ensuring that we do have np.arrays of the good size for batch_x and batch_y before returning them
-        return transform_batches([batch_x, batch_y, batch_w])
-
-    def read_image(self, type_, i):
-        '''
-        :param i: indice of the image or mask to read.
-        :return image: the loaded image with 8 bit pixels, range of values being [0,288]
-        '''
-
-        # Loading the image using 8-bit pixels (0-255)
-        filename = str(type_) + '_%s.png' % i
-        return imread(self.path / filename, flatten=False, mode='L')
-
-    def prepare_image_mask(self):
         """
-        Loads the image and the mask, and discretizes the mask (and converts it in N dimensions, one for each class).
-        :return: Image (ndarray, (H,W)) and Mask (ndarray, (H,W,C)). C number of classes.
-        """
-
-        # We take the next sample to see
-        indice = self.samples_list.pop(0)
-        self.sample_seen += 1
-
-        if self.preload_all:
-            image, mask = self.loaded_data[str(indice)]
-        else:
-            image = self.read_image('image', indice)
-            mask = self.read_image('mask', indice)
 
-        # Discretization of the mask
-        mask = labellize_mask_2d(mask, self.thresh_indices) # mask intensity float between 0-1
+        if (index + 1) * self.batch_size > len(self.ids):
+            self.batch_size = len(self.ids) - index * self.batch_size
```

Comment:
This line doesn't get hit by any of the tests (see [here](https://coveralls.io/builds/24069316/source?filename=AxonDeepSeg/data_management/input_data.py#L72)), just wondering under what circumstances you think this might be hit, and if we should test for it.